newIDE/app/src/Export/LocalExporters/LocalPreviewLauncher/index.js

```diff
@@ -70,6 +72,7 @@ export default class LocalPreviewLauncher extends React.Component<
 
     const win = new BrowserWindow(this.state.previewBrowserWindowConfig);
     win.loadURL(`file://${this.state.previewGamePath}/index.html`);
+    win.setMenuBarVisibility(this.state.hideMenuBar);
```

Comment:
In previous commit, this line was at line number 74(that means right above the `win.loadURL(.....);`). If I keep it there, this is the error I get after executing `npm run flow`.
 ```
Error ---------------------------------------------------- src/Export/LocalExporters/LocalPreviewLauncher/index.js:75:27

Cannot coerce `this.state.previewGamePath` to string because:
 - null or undefined [1] should not be coerced.
 - null [2] should not be coerced.

   src/Export/LocalExporters/LocalPreviewLauncher/index.js:75:27
   75|     win.loadURL(`file://${this.state.previewGamePath}/index.html`);
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

References:
   src/Export/LocalExporters/LocalPreviewLauncher/index.js:32:20
   32|   previewGamePath: ?string,
                          ^^^^^^^ [1]
   src/Export/LocalExporters/LocalPreviewLauncher/index.js:56:22
   56|     previewGamePath: null,
                            ^^^^ [2]
```
Why do I not get the same error now?

Reply:
This is because Flow consider that the call to `win.setMenuBarVisibility` could potentially modify `this.state` and set the variable to `null`. To avoid this, we usually do:

```
// Store the value in a variable
const { previewGamePath } = this.state;
if (!previewGamePath) return; // Check that it's not null (or empty)

// Call any function now

// Here you're 100% sure that previewGamePath is not null (or empty).
// BUT this.state.previewGamePath could have been modified by the previous lines.
// That's why Flow is triggering a potential error if you try to use it.
win.loadURL(`file://${previewGamePath}/index.html`);
```